### PR TITLE
Add a lang service

### PR DESCRIPTION
**Pull request by asaunier**
as suggested by @tsauerwein at https://github.com/c2corg/v6_ui/pull/525#issuecomment-235191318

### CODE REVIEW

File: c2corg_ui/static/js/langservice.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  return this.langs_;
+};
+
+
+/**
+ * @return {string}
+ */
+app.Lang.prototype.detectLang = function() {
+  return this.cookies_.get('interface_lang') ||
+    this.ngeoGetBrowserLanguage_(this.langs_) || 'fr';
```
Comment (tsauerwein) [inline c2corg_ui/static/js/langservice.js]: Can you do it like in the previous `LangController`: In the constructor get the language (from the cookie/browser/default) and set it on `gettextCatalog` as current language. Then if you need to know the current language you call `langService.getLang()` which returns `gettextCatalog.getCurrentLanguage()`.

Comment (asaunier) [inline c2corg_ui/static/js/langservice.js]: Do you mean something like 030ff9a ?

**Comment by tsauerwein** [comment]
Thanks for updating, that looks good!

File: c2corg_ui/static/js/langservice.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+  // server side.
+  var d = new Date();
+  d.setFullYear(d.getFullYear() + 1); // today + 1 year
+  this.cookies_.put('interface_lang', lang, {
+    'path': '/',
+    'expires': d
+  });
+
+  if (this.appAuthentication_.isAuthenticated()) {
+    this.api_.updatePreferredLanguage(lang);
```
Comment (asaunier) [inline c2corg_ui/static/js/langservice.js]: I feel a bit uncomfortable about this code (ported from the lang controller): for authenticated users an "updatePreferredLanguage" request is made to the API for every single page opened by the user, even though the language has not been changed.

Comment (tsauerwein) [inline c2corg_ui/static/js/langservice.js]: At which point do we load the preferred language from the server?

Comment (asaunier) [inline c2corg_ui/static/js/langservice.js]: I don't remember. @gberaudo what do you think?

Comment (gberaudo) [inline c2corg_ui/static/js/langservice.js]: Isn't it retrieved on login?

**Comment by asaunier** [comment]
I have merged the PR. The "updatePreferredLang" issue has been moved at https://github.com/c2corg/v6_ui/issues/540

**Pull request merged**